pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -733,3 +754,42 @@ type forbiddenResponse struct {
 		Code    int    `json:"code"`
 	} `json:"forbidden"`
 }
+
+func (p *provider) cleanupFloatingIP(machine *v1alpha1.Machine, updater cloud.MachineUpdater) error {
+	var floatingIPID string
+	if val, exists := machine.Annotations[floatingIPIDAnnotationKey]; exists {
+		floatingIPID = val
+	} else {
+		return osErrorToTerminalError(fmt.Errorf("failed to release floating ip"),
+			fmt.Sprintf("%s finalizer exists but %s annotation does not", floatingIPReleaseFinalizer, floatingIPIDAnnotationKey))
+	}
+
+	c, _, _, err := p.getConfig(machine.Spec.ProviderConfig)
+	if err != nil {
+		return cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+
+	client, err := getClient(c)
+	if err != nil {
+		return osErrorToTerminalError(err, "failed to get a openstack client")
+	}
+	netClient, err := goopenstack.NewNetworkV2(client, gophercloud.EndpointOpts{Region: c.Region})
+	if err != nil {
+		return fmt.Errorf("failed to create the networkv2 client for region %s: %v", c.Region, err)
+	}
+	if err := osfloatingips.Delete(netClient, floatingIPID).ExtractErr(); err != nil && err.Error() != "Resource not found" {
```

Comment:
Unfortunately gophercloud doesn't provide any way to check for this, nor a const

Reply:
`isNotFoundErr(err)` is in this very package.